File: MtxVec.Delphi/IntroVectors.pas
unit IntroVectors;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  Basic3, StdCtrls, ComCtrls;

type
  TIntroVec = class(TBasicForm3)
    procedure FormCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  IntroVec: TIntroVec;

implementation

{$IFDEF CLR}
{$R *.NFM}
{$ELSE}
{$R *.DFM}
{$ENDIF}

procedure TIntroVec.FormCreate(Sender: TObject);
begin
  inherited;
  With RichEdit1.Lines, RichEdit1 do
  begin
    Clear;
    SelAttributes.Style := [fsBold];
    SelAttributes.Color := RGB(51,104,196);
    SelAttributes.Size := 11;
    Add('Vector operations:');
    Add('');
    Paragraph.FirstIndent := 10;
    Paragraph.Numbering := nsBullet;
    Add('Real and complex FFT');
    Add('Auto and cross correlations');
    Add('Convolution and filtering');
    Add('Complete vector arithmetic');
    Add('Sorting real and complex numbers');
    Add('Random generators');
    Paragraph.FirstIndent := 0;
    Paragraph.Numbering := nsNone;
  end;
end;

initialization
  RegisterClass(TIntroVec);

end.

File: MtxVec.CPP/IntroVectors.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "IntroVectors.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Basic3"
#pragma resource "*.dfm"
TIntroVec *IntroVec;
//---------------------------------------------------------------------------
__fastcall TIntroVec::TIntroVec(TComponent* Owner)
  : TBasicForm3(Owner)
{
//  TBasicForm3::FormCreate(Sender);

  RichEdit1->Lines->Clear();
  RichEdit1->SelAttributes->Style = RichEdit1->SelAttributes->Style << TFontStyle::fsBold;
  RichEdit1->SelAttributes->Color = (TColor) RGB(51,104,196);
  RichEdit1->SelAttributes->Size = 11;
  RichEdit1->Lines->Add("Vector operations:");
  RichEdit1->Lines->Add("");
  RichEdit1->Paragraph->FirstIndent = 10;
  RichEdit1->Paragraph->Numbering = nsBullet;
  RichEdit1->Lines->Add("Real and complex FFT");
  RichEdit1->Lines->Add("Auto and cross correlations");
  RichEdit1->Lines->Add("Convolution and filtering");
  RichEdit1->Lines->Add("Complete vector arithmetic");
  RichEdit1->Lines->Add("Sorting real and complex numbers");
  RichEdit1->Lines->Add("Random generators");
  RichEdit1->Paragraph->FirstIndent = 0;
  RichEdit1->Paragraph->Numbering = nsNone;
}
//---------------------------------------------------------------------------

